src/plugins/index_pattern_management/public/service/column_service/column_service.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IndexPatternTableColumn } from '../../types';
+
+export interface IndexPatternTableColumnServiceSetup {
+  /**
+   * register given column in the registry.
+   */
+  register: (column: IndexPatternTableColumn<unknown>) => void;
+}
+
+export interface IndexPatternTableColumnServiceStart {
+  /**
+   * return all {@link IndexPatternTableColumn | columns} currently registered.
+   */
+  getAll: () => Array<IndexPatternTableColumn<unknown>>;
+}
+
+export class IndexPatternTableColumnService {
+  private readonly columns = new Map<string, IndexPatternTableColumn<unknown>>();
+
+  setup(): IndexPatternTableColumnServiceSetup {
+    return {
+      register: (column) => {
+        if (this.columns.has(column.id)) {
+          throw new Error(`Index Pattern Table Column with id '${column.id}' already exists`);
+        }
```

Comment:
Does this have to `throw` and interrupt the execution or can it be silently ignored? ... or are you handling that elsewhere?

Reply:
Apparently, this is not an edge issue for us to gracefully recover.